src/Keboola/DbExtractor/Application.php

```diff
@@ -8,44 +8,48 @@
 use Keboola\DbExtractor\Configuration\ConfigDefinition;
 use Keboola\DbExtractor\Configuration\ConfigRowDefinition;
 use Keboola\DbExtractor\Exception\UserException;
-use Pimple\Container;
+use Keboola\DbExtractor\Extractor\Extractor;
 use Symfony\Component\Config\Definition\ConfigurationInterface;
 use Symfony\Component\Config\Definition\Exception\Exception as ConfigException;
 use Symfony\Component\Config\Definition\Processor;
 use ErrorException;
 
-class Application extends Container
+class Application
 {
-    /** @var ConfigurationInterface */
-    private $configDefinition;
+    /** @var string */
+    private $action;
 
-    public function __construct(array $config, Logger $logger, array $state = [])
-    {
-        static::setEnvironment();
+    /** @var array */
+    private $parameters;
 
-        parent::__construct();
+    /** @var array */
+    private $state;
 
-        $app = $this;
+    /** @var Logger */
+    private $logger;
 
-        $this['action'] = isset($config['action'])?$config['action']:'run';
+    /** @var Extractor */
+    private $extractor;
 
-        $this['parameters'] = $config['parameters'];
+    /** @var ConfigurationInterface */
+    private $configDefinition;
 
-        $this['state'] = $state;
+    public function __construct(array $config, Logger $logger, array $state = [])
+    {
+        static::setEnvironment();
 
-        $this['logger'] = $logger;
+        $this->action = isset($config['action'])?$config['action']:'run';
+        $this->parameters = $config['parameters'];
+        $this->state = $state;
+        $this->logger = $logger;
 
-        $this['extractor_factory'] = function () use ($app) {
-            return new ExtractorFactory($app['parameters'], $app['state']);
-        };
+        $extractorFactory = new ExtractorFactory($this->parameters, $this->state);
+        $this->extractor = $extractorFactory->create($this->logger);
 
-        $this['extractor'] = function () use ($app) {
-            return $app['extractor_factory']->create($app['logger']);
-        };
-        if (isset($this['parameters']['tables'])) {
+        if (isset($this->parameters['tables'])) {
```

Comment:
vím, že to není předmětem tohohle PR, ale jaký má tohle smysl? Je to takové atypické - že bys podle configu něco rozhodoval a pak teprve řešil "action"...

Reply:
Resili jsme to tu, take se mi to uplne nelibi, ale zatim si nejsem uplne jisty jestli se predelanim neco nerozbije https://keboola.slack.com/archives/C035LGNK5/p1537949500000100